pkg/internal/erc20/erc20.go

```diff
@@ -0,0 +1,74 @@
+package erc20
+
+import (
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const (
+	UnknownTokenName = "Unknown"
+)
+
+// ABI is a simplified ABI for the ERC20 token standard
+var ABI = `[{"constant":true,"inputs":[],"name":"name","outputs":[{"name":"","type":"string"}],"type":"function"}]`
+
+// ERC20 is the Go binding of the ERC20 contract
+type ERC20 struct {
+	Caller // Read-only binding to the contract
+}
+
+// Caller is an auto generated read-only Go binding around an Ethereum contract.
+type Caller struct {
+	contract *bind.BoundContract // Generic contract wrapper for the low level calls
+}
+
+// Name is a free data retrieval call binding the contract method 0x06fdde03.
+func (c *Caller) Name(opts *bind.CallOpts) (string, error) {
+	var out []interface{}
+	err := c.contract.Call(opts, &out, "name")
+	if err != nil {
+		return "", err
+	}
+	return out[0].(string), err
+}
+
+// NewERC20 creates a new instance of ERC20, bound to a specific deployed contract.
+func NewERC20(address common.Address, backend bind.ContractBackend) (*ERC20, error) {
+	contract, err := bindERC20(address, backend, backend, backend)
+	if err != nil {
+		return nil, err
+	}
+	return &ERC20{Caller: Caller{contract: contract}}, nil
+}
+
+// bindERC20 binds a generic wrapper to an already deployed contract.
+func bindERC20(
+	address common.Address,
+	caller bind.ContractCaller,
+	transactor bind.ContractTransactor,
+	filterer bind.ContractFilterer,
+) (*bind.BoundContract, error) {
+	parsed, err := abi.JSON(strings.NewReader(ABI))
+	if err != nil {
+		return nil, err
+	}
+	return bind.NewBoundContract(address, parsed, caller, transactor, filterer), nil
```

Comment:
nit: return err

Reply:
didn't get this comment?

Reply:
It's inconsistent convention. In some places you're reusing `err` knowing its nil since you've made it past the if-statement check. Other times you are explicitly returning `nil`.

Reply:
sorry I still did not get this. can you give me an example of inconsistency?

Reply:
Haha no worries. When you return, you know err == nil. I saw a place in the code where we return the `err` object rather than specifying the `nil` return. Just pointing out we could reconcile these different uses of convention. It's of course a very small detail.

Reply:
oh I see what you are saying 
`return bind.NewBoundContract(address, parsed, caller, transactor, filterer), nil` 

you want me to return
`return bind.NewBoundContract(address, parsed, caller, transactor, filterer), err`

I think this is the right pattern since I can explicitly see it's nil, if I pass err then you have to see the above code to see what is the err.

Reply:
can you share where I have returned `err` when I know it's nil? probably that needs to be changed

Reply:
ah I see. stupid generated code. I generally don't do it - let me fix it.

Reply:
Agreed on the explicit nil 👍